Framework/DataHandling/src/LoadILLIndirect2.cpp

```diff
@@ -103,24 +111,30 @@ void LoadILLIndirect2::exec() {
   const std::string instrumentPath =
       m_loader.findInstrumentNexusPath(firstEntry);
   setInstrumentName(firstEntry, instrumentPath);
+
   initWorkSpace();
   progress.report("Initialised the workspace");
 
   loadNexusEntriesIntoProperties(filenameData);
   progress.report("Loaded data details");
 
-  loadDataIntoTheWorkSpace(firstEntry);
+  if (m_loadOption == "Diffractometer") {
+    loadDiffractionData(firstEntry);
+  } else {
+    loadDataIntoTheWorkSpace(firstEntry);
+  }
   progress.report("Loaded the data");
 
   runLoadInstrument();
   progress.report("Loaded the instrument");
 
-  moveSingleDetectors(firstEntry);
-  progress.report("Loaded the single detectors");
-
-  rotateTubes();
-  progress.report("Rotating tubes if necessary");
+  if (m_loadOption == "Spectrometer") {
+    moveSingleDetectors(firstEntry);
+    progress.report("Loaded the single detectors");
```

Comment:
These two reports are now only if the option is spectrometer; this is creating a problem if you look at the progress bar. The number of process reports you declare in the beginning must be equal to the number of time you report. Can you fix that in the beginning?